lib/src/Uncertainty/Algorithm/Simulation/PhysicalSpaceCrossEntropyImportanceSampling.cxx

```diff
@@ -0,0 +1,228 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief PhysicalSpaceCrossEntropyImportanceSampling implement class for Cross Entropy Importance Sampling algorithms in Physical Space
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/PhysicalSpaceCrossEntropyImportanceSampling.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/OptimizationProblem.hxx"
+#include "openturns/NLopt.hxx"
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class PhysicalSpaceCrossEntropyImportanceSampling
+ */
+
+CLASSNAMEINIT(PhysicalSpaceCrossEntropyImportanceSampling)
+
+
+static const Factory<PhysicalSpaceCrossEntropyImportanceSampling> Factory_PhysicalSpaceCrossEntropyImportanceSampling;
+
+
+// Default constructor
+PhysicalSpaceCrossEntropyImportanceSampling::PhysicalSpaceCrossEntropyImportanceSampling()
+  : CrossEntropyImportanceSampling()
+{
+  // Nothing TO DO
+}
+
+
+
+PhysicalSpaceCrossEntropyImportanceSampling::PhysicalSpaceCrossEntropyImportanceSampling(const RandomVector & event,
+                                                                                         const Distribution & auxiliaryDistribution,
+                                                                                         const Indices & activeParameters,
+                                                                                         const Point & initialAuxiliaryDistributionParameters,
+                                                                                         const Interval & bounds,
+                                                                                         const Scalar quantileLevel)
+  : CrossEntropyImportanceSampling(event, quantileLevel)
+  , activeParameters_(activeParameters)
+  , solver_(new NLopt("LD_LBFGS"))
```

Comment:
Why do you use a raw pointer here? Use
```
 , solver_(NLopt("LD_LBFGS"))
```
instead.

Reply:
I've copy paste a using in other class. This is now fixed.